public/app/features/admin/UserProfile.tsx

```diff
@@ -16,163 +16,159 @@ interface Props {
   onPasswordChange(password: string): void;
 }
 
-interface State {
-  isLoading: boolean;
-  showDeleteModal: boolean;
-  showDisableModal: boolean;
-}
-
-export class UserProfile extends PureComponent<Props, State> {
-  state = {
-    isLoading: false,
-    showDeleteModal: false,
-    showDisableModal: false,
-  };
+export function UserProfile(props: Props) {
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showDisableModal, setShowDisableModal] = useState(false);
 
-  showDeleteUserModal = (show: boolean) => () => {
-    this.setState({ showDeleteModal: show });
+  const deleteUserRef = useRef<HTMLButtonElement | null>(null);
+  const showDeleteUserModal = (show: boolean) => () => {
+    setShowDeleteModal(show);
+    if (!show && deleteUserRef.current) {
+      deleteUserRef.current.focus();
+    }
   };
 
-  showDisableUserModal = (show: boolean) => () => {
-    this.setState({ showDisableModal: show });
+  const disableUserRef = useRef<HTMLButtonElement | null>(null);
+  const showDisableUserModal = (show: boolean) => () => {
+    setShowDisableModal(show);
+    if (!show && disableUserRef.current) {
+      disableUserRef.current.focus();
+    }
   };
 
-  onUserDelete = () => {
-    const { user, onUserDelete } = this.props;
+  const onUserDelete = () => {
+    const { user, onUserDelete } = props;
```

Comment:
Can we maybe destructure the props already during the function definition so we could avoid redeclaring them in every function call? 
```
export function UserProfile({user, onUserDelete}: Props) {
```